inc/Engine/Admin/HealthCheck.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace WP_Rocket\Engine\Admin;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+class HealthCheck implements Subscriber_Interface {
+	/**
+	 * Options_Data instance
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Options_Data $options Options_Data instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Returns an array of events that this subscriber wants to listen to.
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'admin_notices' => 'missed_cron',
+		];
+	}
+
+	/**
+	 * Display a warning notice if WP Rocket scheduled events are not running properly
+	 *
+	 * @since 3.5.4
+	 *
+	 * @return void
+	 */
+	public function missed_cron() {
+		if ( ! current_user_can( 'rocket_manage_options' ) ) {
+			return;
+		}
+
+		if ( 'settings_page_wprocket' !== get_current_screen()->id ) {
+			return;
+		}
+
+		$dismissed = (array) get_user_meta( get_current_user_id(), 'rocket_boxes', true );
+
+		if ( in_array( 'rocket_warning_cron', $dismissed, true ) ) {
+			return;
+		}
+
+		if (
+				0 === (int) $this->options->get( 'purge_cron_interval', 0 )
+				&& 0 === (int) $this->options->get( 'async_css', 0 )
+				&& 0 === (int) $this->options->get( 'manual_preload', 0 )
+				&& 0 === (int) $this->options->get( 'schedule_automatic_cleanup', 0 )
+			) {
+			return;
+		}
+
+		$events = [
+			'rocket_purge_time_event'                      => 'Scheduled Cache Purge',
+			'rocket_database_optimization_time_event'      => 'Scheduled Database Optimization',
+			'rocket_database_optimization_cron_interval'   => 'Database Optimization Process',
+			'rocket_preload_cron_interval'                 => 'Preload',
+			'rocket_critical_css_generation_cron_interval' => 'Critical Path CSS Generation Process',
+		];
```

Comment:
Shouldn’t we translate the descriptions?
```suggestion
		$events = [
			'rocket_purge_time_event'                      => __( 'Scheduled Cache Purge', 'rocket' ),
			'rocket_database_optimization_time_event'      => __( 'Scheduled Database Optimization', 'rocket' ),
			'rocket_database_optimization_cron_interval'   => __( 'Database Optimization Process', 'rocket' ),
			'rocket_preload_cron_interval'                 => _x( 'Preload', 'noun', 'rocket' ),
			'rocket_critical_css_generation_cron_interval' => __( 'Critical Path CSS Generation Process', 'rocket' ),
		];
```